src/bitswap/unwant.js

```diff
@@ -1,13 +1,33 @@
 'use strict'
 
 const promisify = require('promisify-es6')
+const CID = require('cids')
 
 module.exports = (send) => {
   return promisify((args, opts, callback) => {
     if (typeof (opts) === 'function') {
       callback = opts
       opts = {}
     }
+
+    // Mirrors block.get's parsing of the CID
+    let cid
+    try {
+      if (CID.isCID(args)) {
+        cid = args
+        args = cid.toBaseEncodedString()
+      } else if (Buffer.isBuffer(args)) {
+        cid = new CID(args)
+        args = cid.toBaseEncodedString()
+      } else if (typeof args === 'string') {
+        cid = new CID(args)
+      } else {
+        return callback(new Error('invalid argument'))
+      }
+    } catch (err) {
+      return callback(err)
+    }
```

Comment:
I'm just wondering if there's a reason for the type detection here? It's a duplication of what's [done in the CID constructor](https://github.com/ipld/js-cid/blob/4d7bcf860214fef04700ebe338908d36182c931e/src/index.js#L56-L88). Would it be possible to do this instead?:

```js
let cid

try {
  cid = new CID(args)
} catch (err) {
  return callback(err)
}

send({
  path: 'bitswap/unwant',
  args: cid.toBaseEncodedString(),
  qs: opts
}, callback)
```